Cython/Debugger/libcython.py

```diff
@@ -165,6 +226,45 @@ def __init__(self,
 
 # General purpose classes
 
+def frame_repr(frame):
+    """Returns a string representing the internal state of a provided GDB frame
+
+    Created to serve as GDB.Frame.__repr__ for debugging purposes. GDB has many
+    layers of abstraction separating the state of the debugger from the
+    corresponding source code. This prints a tree of instance properties,
+    expanding the values for Symtab_and_line, Symbol, and Symtab.
+
+    Most of these properties require computation to determine, meaning much of
+    relevant info is behind a monad. Rather than list which properties are worth
+    printing, we assume that frames are mostly a read-only context and that any
+    function without an argument is idempotent. That is to say, every function
+    is called without arguments and any that don't throw an exception are added
+    to the output.
+
+    Arguments
+    frame       The GDB.Frame instance to be represented as a string
+    """
+    res = str(frame) + "\n"
+    for attribute in sorted(dir(frame)):
+        if attribute.startswith("__"):
+            continue
+        value = getattr(frame, attribute)
+        if callable(value):
+            try:
+                value = value()
+            except Exception as e:
+                pass
```

Comment:
Calling arbitrary methods on frame objects seems unexpected and risky if the goal is something as obviously read-only as printing out their state. I read the docstring and your reasoning, but it doesn't really convince me that this is a good idea. Frame objects have been subject to major changes recently and in the past, so we should try to make little assumptions about their behaviour.

I'd rather see a whitelist of safe and relevant properties to be printed. This is only for printing, after all. If users want more information, they can query it.